src/lib/y2network/config_reader/sysconfig_routes_reader.rb

```diff
@@ -114,9 +118,73 @@ def build_route(hash)
         Y2Network::Route.new(
           to:        hash["destination"] != DEFAULT_DEST ? build_ip(hash["destination"], mask) : :default,
           interface: iface,
-          gateway:   build_ip(hash["gateway"], MISSING_VALUE)
+          gateway:   build_ip(hash["gateway"], MISSING_VALUE),
+          options:   hash["extrapara"] || ""
         )
       end
+
+      # SCR agent for routes files definition
+      def ifroute_term(path)
+        raise ArgumentError if path.nil? || path.empty?
+
+        non_empty_str_term = Yast::term(:String, "^ \t\n")
+        whitespace_term = Yast::term(:Whitespace)
+        optional_whitespace_term = Yast::term(:Optional, whitespace_term)
+        routes_content_term = Yast::term(
+          :List,
+          Yast::term(
+            :Tuple,
+            Yast::term(
+              :destination,
+              non_empty_str_term
+            ),
+            whitespace_term,
+            Yast::term(:gateway, non_empty_str_term),
+            whitespace_term,
+            Yast::term(:netmask, non_empty_str_term),
+            optional_whitespace_term,
+            Yast::term(
+              :Optional,
+              Yast::term(:device, non_empty_str_term)
+            ),
+            optional_whitespace_term,
+            Yast::term(
+              :Optional,
+              Yast::term(
+                :extrapara,
+                Yast::term(:String, "^\n")
+              )
+            )
+          ),
+          "\n"
+        )
+
+        Yast::term(
+          :ag_anyagent,
+          Yast::term(
+            :Description,
+            Yast::term(:File, path),
+            "#\n",
+            false,
+            routes_content_term
+          )
+        )
+      end
+
+      # Registers SCR agent which is used for accessing particular ifroute-device
+      # file
+      #
+      # @param device [String] full path to a file in routes format (e.g. /etc/sysconfig/network/ifroute-eth0)
+      # @return [Path] SCR path of the agent
+      # @raise  [RuntimeError] if it fails
+      def register_ifroute_agent_for_path(path)
+        # /etc/sysconfig/network/ifroute-eth0 define .ifroute-eth0 agent
+        # collisions not handled
+        scr_path = Yast::Path.new(".#{File.basename(path)}")
+        Yast::SCR.RegisterAgent(scr_path, ifroute_term(path)) ||
```

Comment:
Almost four years in the team and I am still learning new things ;-)

Reply:
yeah ... isn't our job beautiful? ;-)